src/Sentry.Extensions.Logging/SentryLogger.cs

```diff
@@ -43,7 +45,9 @@ public bool IsEnabled(LogLevel logLevel)
             return;
         }
 
-        var message = formatter?.Invoke(state, exception);
+        var message = _options.SupportObjectDestructuring
+            ? _sentryLoggerFormatter.Invoke(state)
+            : formatter?.Invoke(state, exception);
```

Comment:
Per our discussion, we can avoid having a custom formatter by creating new `state` instead, then call the original formatter.